src/UI/Implementation/Render/AbstractComponentRenderer.php

```diff
@@ -101,7 +106,11 @@ public function registerResources(ResourceRegistry $registry) {
 	 * @return	string|null
 	 */
 	final protected function bindJavaScript(JavaScriptBindable $component) {
-		$binder = $component->getOnLoadCode();
+        $hash = spl_object_hash($component);
+        if (isset(self::$cache[$hash])) {
+            return self::$cache[$hash];
+        }
```

Comment:
It makes sure that the same ID gets returned if ```Renderer::bindJavaScript()``` is called multiple times with with the same component. E.g.
```php
$id = $this->bindJavaScript($myButton);
$i2 = $this->bindJavaScript($myButton); // $id2 is different than $id1
```
For the current solution, I had to know the ID of the modal when rendering the button. But then the renderer of the modal itself crated a new ID again, so it was not possible to connect the two components.